torch/csrc/jit/python_ivalue.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+#include <pybind11/pybind11.h>
+#include <torch/csrc/python_headers.h>
+
+namespace py = pybind11;
+
+namespace c10 {
+namespace ivalue {
+
+// concrete ivalue Holder that hold a PyObject*
+struct C10_EXPORT ConcretePyObjectHolder final : PyObjectHolder {
+ public:
+  static c10::intrusive_ptr<PyObjectHolder> steal(PyObject* py_obj) {
```

Comment:
Since this already include pybind11, this can just take and store py::object, which would remove all the need for any memory management. The problem before was that the holder was not abstract.